src/chunkserver/masterconn.cc

```
@@ -321,7 +322,13 @@ void masterconn_delete(masterconn */*eptr*/, const std::vector<uint8_t>& data) {
 	matocs::deleteChunk::deserialize(data, chunkId, chunkType, chunkVersion);
 	OutputPacket* outputPacket = new OutputPacket;
 	cstoma::deleteChunk::serialize(outputPacket->packet, chunkId, chunkType, 0);
-	job_delete(jpool, masterconn_saujobfinished, outputPacket, chunkId, chunkVersion, chunkType);
+	if (jobPool) {
+		job_delete(*jobPool, masterconn_saujobfinished, outputPacket, chunkId, chunkVersion,
+		           chunkType);
+	} else {
+		safs::log_err("masterconn_delete: jobPool is null.");
+		delete outputPacket;
+	}
```

What scenarios do you think where jobPool would be null? It would seem something has gone terribly wrong if it's null, but I'm not sure.

---

Although unlikely, I'm thinking in the scenario where the pipe creation fails.
In this case, an exception is thrown and `jobPool` instance is not created correctly.
For sure, there are other scenarios, but this is the first that comes to my mind :)

---

I wonder whether it's better to throw an exception or not. Either way I'd set the error level to critical.

---

Maybe in the future we can change it, as well as the error level. However, I don't think these are stoppers to merge the changes :)